draft-ietf-tls-esni.md

```diff
@@ -719,10 +719,23 @@ Upon receiving an "encrypted_server_name" extension, the client-facing
 server MUST check that it is able to negotiate TLS 1.3 or greater. If not,
 it MUST abort the connection with a "handshake_failure" alert.
 
-If the ClientEncryptedSNI.record_digest value does not match the
-cryptographic hash of any known ESNIKeys structure, it MUST ignore the
-extension and proceed with the connection, with the following added
-behavior:
+The ClientEncryptedSNI value is said to match a known ESNIKeys if there exists
+an ESNIKeys that can be used to successfully decrypt ClientEncryptedSNI.encrypted_sni.
+This matching procedure should be done using one of the following two checks:
+
+1. Compare ClientEncryptedSNI.record_digest against cryptographic hashes of known ESNIKeys 
+and choose the one that matches. 
+2. Use trial decryption of ClientEncryptedSNI.encrypted_sni with known ESNIKeys and choose 
+the one that succeeds.
+
+Which check to use must be configured externally. Some uses of ESNI, such as local 
```

Comment:
I would write this as:
"Unless specified by the application using DTLS or externally configured on both sides, implementations MUST use the first method."

Reply:
That's much better -- thanks!